XiEditor/EditViewController.swift

```diff
@@ -78,9 +78,14 @@ class EditViewController: NSViewController, EditViewDataSource {
         editView.dataSource = self
         gutterView.dataSource = self
         scrollView.contentView.documentCursor = NSCursor.iBeam();
-        
+    }
+
+    override func viewDidAppear() {
+        super.viewDidAppear()
         NotificationCenter.default.addObserver(self, selector: #selector(EditViewController.boundsDidChangeNotification(_:)), name: NSNotification.Name.NSViewBoundsDidChange, object: scrollView.contentView)
         NotificationCenter.default.addObserver(self, selector: #selector(EditViewController.frameDidChangeNotification(_:)), name: NSNotification.Name.NSViewFrameDidChange, object: scrollView)
+        // call to set initial scroll position once we know view size
+        updateEditViewScroll()
```

Comment:
by moving this to `viewDidAppear` the update fires, because the view's bounds have been set. There were two previous problems: the first was that sometimes the document wasn't set, but the other one was that the notification was sent with the height of the view as set in interface builder (274pt or something) and not the height of the correctly resized view.

I played around with this change and it was working consistently. If it turns out there _is_ a race condition around the setting of the document (and I don't think there should be, the call order of lifecycle methods should be deterministic?) than we will see crashes and I'll try a different approach.